src/javascript-wrapper.js

```diff
@@ -64,26 +36,19 @@ var fRunUpdater = function(updater, updaterFormals, vni) {
      }).then(function(results) { 
          // console.log('updater returned results: ',results);
 
-         if (! _.isUndefined(results)) { 
+         if (! _.isUndefined(results)) {
              // Got some results back from updater
              // If the updater returned anything, set the output state with it
-             var newState = vni.outputState();
-             if (newState.lm === oldOutputStateLm || 
-                 _.isUndefined(oldOutputStateLm) && ! _.isUndefined(newState.lm)) { 
+             var newStateLm = vni.outputState().lm;
+             if (newStateLm === oldOutputStateLm ||
+                 _.isUndefined(oldOutputStateLm) && ! _.isUndefined(newStateLm)) {
```

Comment:
That check is there to deal with the first time we execute a component's updater.  In that case, the original OutputStateLM will be undefined since there has been no output yet. The newStateLM (which is really an output state lm), is going to have it's first LM value since we just got it above.   In that case, we do want to pass the data downstream, no?